pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -297,6 +305,94 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 	return ret, nil
 }
 
+// newRBDSnapshotClass returns a new RBD snapshotter backed SnapshotClass
+func (r *ReconcileStorageCluster) newRBDSnapshotClass(initData *ocsv1.StorageCluster) *snapapi.VolumeSnapshotClass {
+	retSC := &snapapi.VolumeSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: generateNameForSnapshotClass(initData, "rbd"),
+		},
+		Driver: generateNameForSnapshotClassDriver(initData, "rbd"),
+		Parameters: map[string]string{
+			"clusterID":                initData.Namespace,
+			snapshotterSecretName:      "rook-csi-rbd-provisioner",
+			snapshotterSecretNamespace: initData.Namespace,
+		},
+		DeletionPolicy: snapapi.VolumeSnapshotContentDelete,
+	}
+	return retSC
+}
+
+// newCephFsSnapshotClass returns a new CephFs snapshotter backed SnapshotClass
+func (r *ReconcileStorageCluster) newCephFsSnapshotClass(initData *ocsv1.StorageCluster) *snapapi.VolumeSnapshotClass {
+	retSC := &snapapi.VolumeSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: generateNameForSnapshotClass(initData, "cephfs"),
+		},
+		Driver: generateNameForSnapshotClassDriver(initData, "cephfs"),
+		Parameters: map[string]string{
+			"clusterID":                initData.Namespace,
+			snapshotterNamePrefix:      fmt.Sprintf("%s-%s-", initData.Name, initData.Namespace),
```

Comment:
This is not supported in downstream. need to remove it

Reply:
@aruniiird I would have been happy if we could create a snapshot class based on the arg passed to it and get rid of the duplication.. for example:
` func newSnapshotClass( driverName, type) 
or to the below format
`func newSnapshotClass(type)
 . then generate the snapshot driver from the type ?`

Reply:
Yup, done...